src/usersyncroniser.ts

```diff
@@ -372,4 +390,14 @@ export class UserSyncroniser {
     private displayNameForUser(discordUser): string {
         return `${discordUser.username}#${discordUser.discriminator}`;
     }
+
+    private async leave(intent: Intent, roomId: string, checkCache: boolean = true) {
+        const userId = intent.getClient().getUserId();
+        if (checkCache && ![null, "join", "invite"]
+            .includes(intent.opts.backingStore.getMembership(roomId, userId))) {
+            return;
+        }
+        await intent.leave(roomId);
+        intent.opts.backingStore.setMembership(roomId, userId, "leave");
```

Comment:
I thought intent.leave would set the membership anyway, or the leave event that comes down the AS stream.

Reply:
soru tested and apparently it doesn't